test/lib/ufe/testParentCmd.py

```diff
@@ -1175,48 +1170,84 @@ def checkAfter(a, b, c, e, f):
         children = hierarchyAfter()
         checkAfter(*children)
 
-    # Test is currently crashing and needs to be investigated
-    def _testEditRouter(self):
+    def testEditRouter(self):
         '''Test edit router functionality.'''
 
         cmds.file(new=True, force=True)
         import mayaUsd_createStageWithNewLayer
 
-        # Create the following hierarchy:
+        psPathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        stage = mayaUsd.lib.GetPrim(psPathStr).GetStage()
+
+        # Create the following layer hierarchy:
+        #
+        # anonymousLayer1
+        #  |_ bSubLayer
+        #  |_ aSubLayer
+        #
+        # Sublayer B is thus higher-priority than A.
+        rootLayerId = stage.GetRootLayer().identifier
+        aSubLayerId = cmds.mayaUsdLayerEditor(rootLayerId, edit=True, addAnonymous="aSubLayer")[0]
+        bSubLayerId = cmds.mayaUsdLayerEditor(rootLayerId, edit=True, addAnonymous="bSubLayer")[0]
+
+        # Create the following hierarchy in lower-priority layer A.
         #
         # ps
         #  |_ A
         #      |_ B
         #  |_ C
         #
-        # We select B and C, in order, and parent.  This parents B to C.
-
-        psPathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
-        stage = mayaUsd.lib.GetPrim(psPathStr).GetStage()
+        cmds.mayaUsdEditTarget(psPathStr, edit=True, editTarget=aSubLayerId)
         stage.DefinePrim('/A', 'Xform')
         stage.DefinePrim('/A/B', 'Xform')
         stage.DefinePrim('/C', 'Xform')
 
-        psPath = ufe.PathString.path(psPathStr)
-        psPathSegment = psPath.segments[0]
-        aPath = ufe.Path([psPathSegment, usdUtils.createUfePathSegment('/A')])
-        a = ufe.Hierarchy.createItem(aPath)
-        bPath = aPath + ufe.PathComponent('B')
-        b = ufe.Hierarchy.createItem(bPath)
-        cPath = ufe.Path([psPathSegment, usdUtils.createUfePathSegment('/C')])
-        c = ufe.Hierarchy.createItem(cPath)
-
-        # Add a sub-layer, where the parent edit should write to.
-        subLayerId = cmds.mayaUsdLayerEditor(stage.GetRootLayer().identifier, edit=True, addAnonymous="aSubLayer")[0]
-
+        def firstSubLayer(context, routingData):
+            # Write edits to the highest-priority child layer of the root.
+
+            # Here, prim is the parent prim.
+            prim = context.get('prim')
+            self.assertIsNot(prim, None)
+            self.assertFalse(len(prim.GetStage().GetRootLayer().subLayerPaths)==0)
+            layerId = prim.GetStage().GetRootLayer().subLayerPaths[0]
+            layer = Sdf.Layer.Find(layerId)
+            # Make sure the destination exists in the target layer, otherwise
+            # SdfCopySpec will error.
+            Sdf.JustCreatePrimInLayer(layer, prim.GetPath())
```

Comment:
This is the crucial change in the test: the destination layer is completely empty, but SdfCopySpec requires a destination prim.  Just create an empty prim in the destination layer.